akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala

```diff
@@ -108,27 +104,28 @@ private[remote] abstract class InboundCompression[T >: Null](
    * @throws UnknownCompressedIdException if given id is not known, this may indicate a bug – such situation should not happen.
    */
   // not tailrec because we allow special casing in sub-class, however recursion is always at most 1 level deep
-  def decompress(tableVersion: Long, idx: Int): OptionVal[T] = {
-    val activeVersion = activeTable.version // TODO move into state
+  def decompress(incomingTableVersion: Long, idx: Int): OptionVal[T] = {
+    val activeVersion = activeTable.version
 
-    if (tableVersion == -1) OptionVal.None // no compression, bail out early
-    else if (tableVersion == activeVersion) {
+    if (incomingTableVersion == -1) OptionVal.None // no compression, bail out early
+    else if (incomingTableVersion == activeVersion) {
       val value: T = activeTable.get(idx)
-      if (settings.debug) log.debug(s"Decompress [{}] => {}", idx, value)
       if (value != null) OptionVal.Some[T](value)
       else throw new UnknownCompressedIdException(idx)
-    } else if (tableVersion < activeVersion) {
-      log.debug("Received value compressed with old table: [{}], current table version is: [{}]", tableVersion, activeVersion)
+    } else if (incomingTableVersion < activeVersion) {
+      log.warning("Received value compressed with old table: [{}], current table version is: [{}]", incomingTableVersion, activeVersion)
       OptionVal.None
-    } else if (tableVersion == nextTable.version) {
+    } else if (incomingTableVersion == nextTable.version) {
       advertisementInProgress = false
       log.debug("Received first value compressed using the next prepared compression table, flipping to it (version: {})", nextTable.version)
       startUsingNextTable()
-      decompress(tableVersion, idx) // recurse, activeTable will not be able to handle this
+      decompress(incomingTableVersion, idx) // recurse, activeTable will not be able to handle this
     } else {
       // which means that incoming version was > nextTable.version, which likely is a bug
-      log.error("Inbound message is using compression table version higher than the highest allocated table on this node. " +
-        "This should not happen! State: activeTable: {}, nextTable, incoming tableVersion: {}", activeVersion, nextTable, tableVersion)
+      log.error(
+        "Inbound message is using compression table version higher than the highest allocated table on this node. " +
+          "This should not happen! State: activeTable: {}, nextTable: {}, incoming tableVersion: {}",
+        activeVersion, nextTable.version, incomingTableVersion)
```

Comment:
This can happen when a destination system restarts, and suddenly receives a message from a system that was using compression to this destination already, we drop/fail messages via returning None, but we could fail explicitly here as well. 

The sending side is expected to drop the table (as new handshake happens)